driver/sum_driver.hpp

```diff
@@ -0,0 +1,334 @@
+    inflags.AddInputFlag("InputDimLengths",
+                         'D',
+                         "256,4,8732",
+                         "The dimensional lengths of the input tensor(Default=256,4,8732)",
+                         "string");
+    inflags.AddInputFlag(
+        "DimToReduce", 'R', "1", "The indice of the dimensions to be reduced(Default=1)", "int");
+    inflags.AddInputFlag("NanPropagation",
+                         'N',
+                         "0",
+                         "Nan number propagation mode (check the miopenSumNanPropagation_t in "
+                         "miopen.h) (Default=0 to indicate no Nan propagation)",
+                         "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+std::vector<int> SumDriver<Tgpu, Tref>::GetInputTensorLengthsFromCmdLine()
+{
+    std::string lengthsStr = inflags.GetValueStr("InputDimLengths");
+
+    std::vector<int> lengths;
+    std::size_t pos = 0;
+    std::size_t new_pos;
+
+    new_pos = lengthsStr.find(',', pos);
+    while(new_pos != std::string::npos)
+    {
+        std::string sliceStr = lengthsStr.substr(pos, new_pos - pos);
+
+        int len = std::stoi(sliceStr);
+
+        lengths.push_back(len);
+
+        pos     = new_pos + 1;
+        new_pos = lengthsStr.find(',', pos);
+    };
+
+    std::string sliceStr = lengthsStr.substr(pos);
+    int len              = std::stoi(sliceStr);
+
+    lengths.push_back(len);
+
+    return (lengths);
+}
+
+template <typename Tgpu, typename Tref>
+int SumDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t in_sz  = GetTensorSize(inputDesc);
+    size_t out_sz = GetTensorSize(outputDesc);
+
+    miopenGetSumWorkspaceSize(GetHandle(), inputDesc, dim, outputDesc, &ws_sizeInBytes);
+
+    uint32_t ctx = 0;
+
+    in_dev        = std::unique_ptr<GPUMem>(new GPUMem(ctx, in_sz, sizeof(Tgpu)));
+    out_dev       = std::unique_ptr<GPUMem>(new GPUMem(ctx, out_sz, sizeof(Tgpu)));
+    workspace_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, ws_sizeInBytes, sizeof(Tgpu)));
+
+    in      = std::vector<Tgpu>(in_sz, static_cast<Tgpu>(0));
+    out     = std::vector<Tgpu>(out_sz, static_cast<Tgpu>(0));
+    outhost = std::vector<Tref>(out_sz, static_cast<Tref>(0));
+
+    int status;
+
+    for(int i = 0; i < in_sz; i++)
+    {
+        in[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(0.0), static_cast<Tgpu>(1.0));
+    }
+    status = in_dev->ToGPU(q, in.data());
+
+    status |= out_dev->ToGPU(q, out.data());
+
+    if(status != 0)
+        std::cout << "Error copying data to GPU\n" << std::endl;
```

Comment:
```suggestion
        std::cerr << "Error copying data to GPU" << std::endl;
```
[R] If easy to do, it would be nice to print src and dest addresses and data size. Up to you.

Reply:
fixed it